### PR TITLE
[Dataset] GNNBenchmarkDataset

### PR DESCRIPTION
**Pull request by HuXiangkun**
## Description
#1776 

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented
- [x] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR

### CODE REVIEW

File: python/dgl/data/gnn_benckmark.py [hunk 1]
```diff
@@ -61,68 +93,349 @@ def load_npz(self, file_name):
...
+        Default: ~/.dgl/
+    force_reload : bool
+        Whether to reload the dataset. Default: False
+    verbose: bool
+        Whether to print out progress information. Default: True.
+
+    Attributes
+    ----------
+    num_classes : int
+        Number of classes for each node.
```
Comment (classicsong) [inline python/dgl/data/gnn_benckmark.py:168]: data is still an attribute

File: python/dgl/data/gnn_benckmark.py [hunk 2]
```diff
@@ -61,68 +93,349 @@ def load_npz(self, file_name):
...
```
Comment (classicsong) [inline python/dgl/data/gnn_benckmark.py:225]: data is still an attribute

File: python/dgl/data/gnn_benckmark.py [hunk 3]
```diff
@@ -61,68 +93,349 @@ def load_npz(self, file_name):
...
+        Default: ~/.dgl/
+    force_reload : bool
+        Whether to reload the dataset. Default: False
+    verbose: bool
+        Whether to print out progress information. Default: True.
 
+    Attributes
+    ----------
+    num_classes : int
+        Number of classes for each node.
```
Comment (classicsong) [inline python/dgl/data/gnn_benckmark.py:282]: data is still an attribute

File: python/dgl/data/gnn_benckmark.py [hunk 4]
```diff
@@ -61,68 +93,349 @@ def load_npz(self, file_name):
...
+        Default: ~/.dgl/
+    force_reload : bool
+        Whether to reload the dataset. Default: False
+    verbose: bool
+        Whether to print out progress information. Default: True.
+
+    Attributes
+    ----------
+    num_classes : int
+        Number of classes for each node.
```
Comment (classicsong) [inline python/dgl/data/gnn_benckmark.py:339]: data is still an attribute

File: python/dgl/data/gnn_benckmark.py [hunk 5]
```diff
@@ -61,68 +93,349 @@ def load_npz(self, file_name):
...
```
Comment (classicsong) [inline python/dgl/data/gnn_benckmark.py:394]: data is still an attribute

File: python/dgl/data/gnn_benckmark.py [hunk 6]
```diff
@@ -61,68 +93,349 @@ def load_npz(self, file_name):
...
+
+def Coauthor(name):
+    if name == 'cs':
+        deprecate_class('Coauthor', 'CoauthorCSDataset')
+        return CoauthorCSDataset()
+    elif name == 'physics':
+        deprecate_class('Coauthor', 'CoauthorPhysicsDataset')
+        return CoauthorPhysicsDataset()
+    else:
+        raise ValueError('Dataset name should be "cs" or "physics".')
```
Comment (classicsong) [inline python/dgl/data/gnn_benckmark.py:441]: Do we need to update some examples ?

File: python/dgl/data/gnn_benckmark.py [hunk 7]
```diff
@@ -61,68 +93,349 @@ def load_npz(self, file_name):
...
+
+def AmazonCoBuy(name):
+    if name == 'computers':
+        deprecate_class('AmazonCoBuy', 'AmazonCoBuyComputerDataset')
+        return AmazonCoBuyComputerDataset()
+    elif name == 'photo':
+        deprecate_class('AmazonCoBuy', 'AmazonCoBuyPhotoDataset')
+        return AmazonCoBuyPhotoDataset()
+    else:
+        raise ValueError('Dataset name should be "computers" or "photo".')
```
Comment (hetong007) [inline python/dgl/data/gnn_benckmark.py:430]: Do we need to add docstrings for these raised errors?

Comment (HuXiangkun) [reply python/dgl/data/gnn_benckmark.py:430]: These deprecated class will not appear in API doc.

File: python/dgl/data/gnn_benckmark.py (hunk 6 above)
Comment (HuXiangkun) [reply python/dgl/data/gnn_benckmark.py:441]: There are no examples or tutorials using these datasets.

**Pull request merged**